flink-core/src/main/java/org/apache/flink/api/common/io/FileInputFormat.java

```diff
@@ -706,6 +700,9 @@ public void open(FileInputSplit fileSplit) throws IOException {
 			this.stream = isot.waitForCompletion();
 			this.stream = decorateInputStream(this.stream, fileSplit);
 		}
+		catch (FileNotFoundException e) {
+			throw new FileNotFoundException("Input split " + fileSplit.getPath() + " doesn't exist, skip and continue");
+		}
```

Comment:
I would recommend including the text and stack of the caught ex, for the better stack trace. FNFEs can get raised in odd circumstances in S3; seeing the full stack is what you need when fielding support calls. eg.

```java
throw new FileNotFoundException("Input split " + fileSplit.getPath() + " doesn't exist, skip and continue: " + e, e);
```